aospy/test/test_data_loader.py

```diff
@@ -0,0 +1,392 @@
+            da.copy(), **self.generate_file_set_args)[TIME_STR]
+
+        expected = da[TIME_STR] + np.timedelta64(-3, 'h')
+        expected[TIME_STR] = expected
+        assert result.identical(expected)
+
+        # Test case where intvl_in doesn't end in 'hr'
+        self.generate_file_set_args['intvl_in'] = 'daily'
+        da = inst_ds[self.var_name]
+        result = self.DataLoader._maybe_apply_time_shift(
+            da.copy(), **self.generate_file_set_args)[TIME_STR]
+
+        expected = da[TIME_STR]
+        expected[TIME_STR] = expected
+        assert result.identical(expected)
+
+    def test_maybe_apply_time_offset_ts(self):
+        # Should provide no offset by default
+        ds = xr.decode_cf(self.ds)
+        da = ds[self.var_name]
+
+        # If you don't pass a time_offset or non-null DataVars,
+        # nothing should happen
+        result = self.DataLoader._maybe_apply_time_shift(
+            da.copy(), **self.generate_file_set_args)[TIME_STR]
+        assert result.identical(da[TIME_STR])
+
+    def test_generate_file_set(self):
+        with self.assertRaises(IOError):
+            self.DataLoader._generate_file_set(**self.generate_file_set_args)
+
+    def test_input_data_paths_gfdl(self):
+        expected = ['/test/atmos/ts/monthly/6yr/atmos.200601-201112.temp.nc']
+        result = self.DataLoader._input_data_paths_gfdl(
+            'temp', datetime(2010, 1, 1), datetime(2010, 12, 31), 'atmos',
+            'monthly', 'pressure', 'ts', None)
+        self.assertEqual(result, expected)
+
+        expected = ['/test/atmos_daily/ts/daily/6yr/'
+                    'atmos_daily.20060101-20111231.temp.nc']
+        result = self.DataLoader._input_data_paths_gfdl(
+            'temp', datetime(2010, 1, 1), datetime(2010, 12, 31), 'atmos',
+            'daily', 'pressure', 'ts', None)
+        self.assertEqual(result, expected)
+
+        expected = ['/test/atmos_level/ts/monthly/'
+                    '6yr/atmos_level.200601-201112.temp.nc']
+        result = self.DataLoader._input_data_paths_gfdl(
+            'temp', datetime(2010, 1, 1), datetime(2010, 12, 31), 'atmos',
+            'monthly', ETA_STR, 'ts', None)
+        self.assertEqual(result, expected)
+
+        expected = ['/test/atmos/ts/monthly/'
+                    '6yr/atmos.200601-201112.ps.nc']
+        result = self.DataLoader._input_data_paths_gfdl(
+            'ps', datetime(2010, 1, 1), datetime(2010, 12, 31), 'atmos',
+            'monthly', ETA_STR, 'ts', None)
+        self.assertEqual(result, expected)
+
+        expected = ['/test/atmos_inst/ts/monthly/'
+                    '6yr/atmos_inst.200601-201112.temp.nc']
+        result = self.DataLoader._input_data_paths_gfdl(
+            'temp', datetime(2010, 1, 1), datetime(2010, 12, 31), 'atmos',
+            'monthly', 'pressure', 'inst', None)
+        self.assertEqual(result, expected)
+
+        # 2017-01-16 SKC: this test fails; I'm not sure what the result
```

Comment:
This is reflective of my ignorance of all the possible use cases enabled by the GFDLDataLoader (I've only used it a few times a while back).  Could you point me to an example for when ```'monthly_from_'``` is in ```dtype_in_time```?  I was honestly just guessing here.